numpy/lib/function_base.py

```diff
@@ -75,6 +76,87 @@ def iterable(y):
     return 1
 
 
+def _hist_optim_numbins_estimator(a, estimator):
+    """
+    A helper function to be called from histogram to deal with estimating optimal number of bins
+
+    if estimator is a string
+        Check the corresponding estimator exists and
+        returns the optimal number of bins calculated using the estimator
+    """
+    if not isinstance(estimator, basestring):
+        raise ValueError("bins = '{0}', should be a string".format(estimator))
```

Comment:
this method is only called privately, so (personally) I would not bother with raising a ValueError but rather write `assert isinstance(estimator, basestring)`